File: output/xCodeEval_remove_8000/C++/Java/640dc614f856ecc5cfc468b141ab77d4.java
import java.util.*;
import java.io.*;
import java.math.*;
import java.text.*;
import java.time.*;

class Main {
    static final long Mod = 1000000007L, INF = 1_000_000_000L, LINF = 1_000_000_000_000_000_000L;
    static final double Pi = 3.141592653589793116, EPS = 1e-9, Gold = ((1 + Math.sqrt(5)) / 2);
    static final long[] keymod = {1000000007L, 1000000009L, 1000000021L, 1000000033L};
    static final int keyCount = keymod.length;

    static long getbit(long s, int i) {
        return (s >> i) & 1;
    }

    static long onbit(long s, int i) {
        return s | (1L << i);
    }

    static long offbit(long s, int i) {
        return s & (~(1L << i));
    }

    static int cntbit(long s) {
        return Long.bitCount(s);
    }

    static long x;

    static class Pair {
        long first, second;

        Pair(long first, long second) {
            this.first = first;
            this.second = second;
        }
    }

    static Pair func(long m, long n, long s) {
        if (m == 0) return new Pair(n, s);
        long k = (long) Math.cbrt(m);
        long x = k * k * k, y = (k - 1) * (k - 1) * (k - 1);
        return max(func(m - x, n + 1, s + x), func(x - y - 1, n + 1, s + y));
    }

    static void input() {
        Scanner sc = new Scanner(System.in);
        x = sc.nextLong();
        sc.close();
    }

    static void solve() {
        Pair token = func(x, 0, 0);
        System.out.println(token.first + " " + token.second);
    }

    public static void main(String[] args) {
        input();
        solve();
    }
}

File: output/GPT3.5-Turbo/xCodeEval_remove_4000/C++/Go/640dc614f856ecc5cfc468b141ab77d4.go
package main

import (
	"fmt"
	"math"
	"time"
)

type pii struct {
	first  int64
	second int64
}

var (
	Mod     int64 = 1000000007
	INF     int64 = 1e9
	LINF    int64 = 1e18
	Pi      float64 = 3.141592653589793116
	EPS     float64 = 1e-9
	Gold    float64 = (1 + math.Sqrt(5)) / 2
	keymod         = []int64{1000000007, 1000000009, 1000000021, 1000000033}
	keyCount       = len(keymod)
	TimeStart      time.Time
	TimeEnd        time.Time
)

func getbit(s int64, i int) int {
	return int((s >> i) & 1)
}

func onbit(s int64, i int) int64 {
	return s | (1 << i)
}

func offbit(s int64, i int) int64 {
	return s &^ (1 << i)
}

func cntbit(s int64) int {
	return int(math.Popcount(s))
}

func func(m, n, s int64) pii {
	if m == 0 {
		return pii{n, s}
	}
	k := int64(math.Cbrt(float64(m)))
	x := k * k * k
	y := (k - 1) * (k - 1) * (k - 1)
	return max(func(m-x, n+1, s+x), func(x-y-1, n+1, s+y))
}

func max(a, b pii) pii {
	if a.first > b.first {
		return a
	} else if a.first < b.first {
		return b
	} else {
		if a.second > b.second {
			return a
		} else {
			return b
		}
	}
}

func Input() int64 {
	var x int64
	fmt.Scan(&x)
	return x
}

func Solve(x int64) {
	Token := func(x, 0, 0)
	fmt.Println(Token.first, Token.second)
}

func main() {
	TimeStart = time.Now()

	x := Input()
	Solve(x)

	var T int
	fmt.Scan(&T)
	for T > 0 {
		x := Input()
		Solve(x)
		T--
	}

	TimeEnd = time.Now()
	ElapsedTime := TimeEnd.Sub(TimeStart)
	fmt.Printf("\n\nTime elapsed: %f seconds.\n", ElapsedTime.Seconds())
}
